addons/project/models/project_task.py

```diff
@@ -938,13 +938,32 @@ def create(self, vals_list):
             # if the portal user could really create the tasks based on the ir rule.
             tasks.with_user(self.env.user).check_access_rule('create')
         current_partner = self.env.user.partner_id
+
+        all_partner_emails = []
+        for task in tasks:
+            all_partner_emails += tools.email_split(task.email_cc)
+        partners = self.env['res.partner'].search([('email', 'in', all_partner_emails)])
+        partner_per_email = {
+            partner.email: partner
+            for partner in partners
+            if not all(u.share for u in partner.user_ids)
+        }
         for task in tasks:
             if task.project_id.privacy_visibility == 'portal':
                 task._portal_ensure_token()
             for follower in task.parent_id.message_follower_ids:
                 task.message_subscribe(follower.partner_id.ids, follower.subtype_ids.ids)
             if current_partner not in task.message_partner_ids:
                 task.message_subscribe(current_partner.ids)
+            if task.email_cc:
+                partners_with_internal_user = self.env['res.partner']
+                for email in tools.email_split(task.email_cc):
+                    if partner_per_email.get(email):
+                        partners_with_internal_user |= partner_per_email.get(email)
```

Comment:
```suggestion
                    if email in partner_per_email:
                        partners_with_internal_user |= partner_per_email[email]
```
For the first line, I guess if there is an email, than there's a partner.
Second one, we know it is in the dict, so no need to get :)